feat: Add authentication to the App

gvgesanto2 opened the pull request:
## Description

**Task:** https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md

This PR makes the following changes:

- Adds authentication to the application using [Devise](https://github.com/heartcombo/devise#getting-started).
- Adds the functionality to sign up a new user.
- Adds the functionality to edit the user information (including changing the password).
- Adds the ``sign in`` functionality.
- Adds the ``sign out`` functionality.

## Type of change

- [ ] Performance (non-breaking change which fixes performance issues)
- [ ] Test (change related to testing)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project (no linters errors)
- [X] I have checked my code and corrected any misspellings
- [X] I have deleted any unused code as comments in my codebase

gvgesanto2 committed:
feat: Set up Devise for authentication

gvgesanto2 committed:
Merge branch 'dev' into auth

gvgesanto2 committed:
Add the sing up, sign in, sign out, and edit account links to the header

gvgesanto2 committed:
fix: Fix linters errors

ShoiraTa commented on app/controllers/application_controller.rb:
```diff
@@ -1,5 +1,16 @@
+
+  protected
+
+  def update_allowed_parameters
+    devise_parameter_sanitizer.permit(:sign_up) do |u|
+      u.permit(:name, :email, :password, :password_confirmation)
+    end
+    devise_parameter_sanitizer.permit(:account_update) do |u|
+      u.permit(:name, :bio, :email, :password, :password_confirmation, :current_password)
+    end
```
- [ ] Good implementation of allowing custom params 👍🏻

ShoiraTa commented on app/models/user.rb:
```diff
@@ -1,4 +1,7 @@
 class User < ApplicationRecord
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
```
- [ ] The application should ask confirmation of email. To do that you have to uncomment `confirmation` in device migration file. When `confirmable` is on, you will get the link in rails console no matter which email you enter. After, you can just copy the url to confirm and paste in browser.

ShoiraTa reviewed (commented):
Hi @gvgesanto2 ,

:thumbsup: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

**First let me to point out:**

- Great commit history! ✔️ 
- You passed all the Linters! ✔️ 
- Professional pull request! ✔️ 

Keep on coding, good job so far! :clap:

There are some issues that you still need to work on to go to the next project but you are almost there!

<img src="https://media.giphy.com/media/dEPGknRiEniec/giphy.gif" width="500" height="250" />

 ## Required Changes  :recycle: 

_Check the comments under the review._ 

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear by mentioning my Github username [@shoirata].


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

gvgesanto2 committed:
feat: Add the email confirmation functionality

gvgesanto2 commented:
## Description

**Task:** https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md

This PR addresses the change requested by the [last review](https://github.com/gvgesanto2/microverse-rails-blog/pull/9#pullrequestreview-1133579869).

## Type of change

- [ ] Performance (non-breaking change which fixes performance issues)
- [ ] Test (change related to testing)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project (no linters errors)
- [X] I have checked my code and corrected any misspellings
- [X] I have deleted any unused code as comments in my codebase

tufoinnkuo10 reviewed (approved):
## Status: Approved :heavy_check_mark:

## Hi @gvgesanto2 

Your project is complete! Great job for making the requested changes 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 


![tufoin](https://media.giphy.com/media/MaJ7An3EUgLCCh4lXS/giphy.gif)


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
